ports/esp32/esp32_rmt.c

```diff
@@ -46,105 +50,101 @@
 // Originally designed to generate infrared remote control signals, the module is very
 // flexible and quite easy-to-use.
 //
-// This current MicroPython implementation lacks some major features, notably receive pulses
-// and carrier output.
-
-// Last available RMT channel that can transmit.
-#define RMT_LAST_TX_CHANNEL (SOC_RMT_TX_CANDIDATES_PER_GROUP - 1)
+// This code exposes the RMT TX feature.
 
 // Forward declaration
 extern const mp_obj_type_t esp32_rmt_type;
 
 typedef struct _esp32_rmt_obj_t {
     mp_obj_base_t base;
-    uint8_t channel_id;
+    rmt_channel_handle_t channel;
+    bool enabled;
     gpio_num_t pin;
-    uint8_t clock_div;
-    mp_uint_t num_items;
-    rmt_item32_t *items;
-    bool loop_en;
+    uint32_t clock_freq;
+    int resolution_hz;
+    mp_uint_t cap_items;
+    rmt_symbol_word_t *items;
+    int loop_count;
+    int tx_ongoing;
```

Comment:
Can `tx_ongoing` be an `unsigned int`?  It seems that it should never be negative.